utils/hardware_info.go

```diff
@@ -0,0 +1,61 @@
+package utils
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"runtime"
+	"strings"
+
+	"go.viam.com/utils"
+)
+
+// GetArchitectureInfo returns the architecture of the board.
+func GetArchitectureInfo(modelName string) (utils.StringSet, error) {
+	arch := runtime.GOARCH
+
+	switch {
+	case strings.HasPrefix(arch, "amd"):
+		return stringSetFromX86(modelName)
+	case strings.HasPrefix(arch, "arm"):
+		return stringSetFromARM(modelName)
+	default:
+		return nil, noBoardError(modelName)
+	}
+}
+
+// A helper function for ARM architecture to process contents of a
+// given content of a file from os.ReadFile.
+func stringSetFromARM(modelName string) (utils.StringSet, error) {
```

Comment:
Comment doesn't match code: this is a helper to find the set of compatible board descriptions for a devicetree system. (I've also phrased this poorly; hopefully you can take what I wrote and make it better.)